s3fs/core.py

```diff
@@ -478,6 +480,39 @@ def info(self, path, version_id=None):
             except ParamValidationError as e:
                 raise ValueError('Failed to head path %r: %s' % (path, e))
         return super().info(path)
+    
+    def checksum(self, path, refresh=False):
+        """
+        Unique value for current version of file
+
+        If the checksum is the same from one moment to another, the contents
+        are guaranteed to be the same. If the checksum changes, the contents
+        *might* have changed.
+
+        Parameters
+        ----------
+        path : string/bytes
+            file or path to get checksum for
+        refresh : bool (=False)
+            if False, look in local cache for file details first
+        
+        """
+
+        listing = self.ls(path, detail=True, refresh=refresh)
+        
+        def get_checksum(item):
```

Comment:
I wouldn't make a dynamic function here; it could be a static one.
This is especially true if the method only works on one file at a time.